tangelo/toolboxes/molecular_computation/fno.py

```diff
@@ -0,0 +1,413 @@
+# Copyright 2023 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module containing datastructures for interfacing with the Frozen Natural
+Orbitals (FNOs), to automatically truncate the virtual orbital space.
+
+Reference: arXiv:2002.07901
+"""
+
+import warnings
+
+import numpy as np
+
+from tangelo.algorithms.classical import MP2Solver
+
+
+class FNO:
+    """Class to interface with the Frozen Natural Orbitals protocol, that aims
+    at reducing the computational cost of a molecular problem by truncating
+    the virtual space. In general, the virtual orbitals are ranked according to
+    their MP2 occupancies, and selected with a given threshold. They are also
+    transformed to the FNO basis, using the eigenvectors of the MP2
+    virtual-virtual density matrix.
+
+    Attributes:
+            sqmol (SecondQuantizedMolecule): Self-explanatory.
+            uhf (bool): Flag indicating the type of mean field used.
+            n_mos (int): Number of molecular orbitals.
+            fock_ao (np.array): Fock matrix in atomic orbital form.
+            frozen_occupied (list): List of indices of frozen occupied orbitals.
+            threshold (float or list): Threshold(s) for FNO occupancy.
+
+        Properties:
```

Comment:
Your indentation here is odd (I would have expected it to be aligned with `Attributes`).

I'm not sure about the doc for properties and methods. 

These will be rendered in the sphinx doc, users can scroll and see them easily (the attributes can be set etc anywhere so its trickier). On top of that you're forcing developers to update your docstring every time a method/property is removed / changed / added.

The description of your methods does not seem very helpful, they are pretty straightforward and its redundant with the method's docstring. Finally, I don't believe this is consistent with the rest of the codebase.

I'm in favour of deleting these two paragraphs, since this is not consistent